src/sparseml/transformers/compression/utils/compress_save.py

```diff
@@ -0,0 +1,130 @@
+    supports compression
+    """
+
+    def save_pretrained_compressed(save_pretrained_method):
+        if getattr(save_pretrained_method, "_overridden", False):
+            # `model.save_pretrained` has already been replaced, return.
+            return save_pretrained_method
+
+        # Keep a weak reference to the model class and unbound save_pretrained
+        # method so we can call the original
+        model_ref = weakref.ref(save_pretrained_method.__self__)
+        original_save_pretrained = save_pretrained_method.__func__
+        model_class = model_ref().__class__
+        del save_pretrained_method
+
+        @wraps(original_save_pretrained)
+        def save_pretrained_wrapper(
+            save_directory: str,
+            sparsity_config: Optional[CompressionConfig] = None,
+            save_compressed: bool = False,
+            skip_compression_stats: bool = False,
+            **kwargs,
+        ):
+            """
+            Wrapper around PreTrainedModel.save_pretrained(), adds functionality for
+            saving models in a compressed format on disk. The compression format is
+            saved to the model's config file
+
+            :param save_directory: output directory to save model to
+            :param sparsity_config: optional sparsity config to compress model with,
+            if no config is provided it will be inferred from the model
+            :param save_compresed: whether or not to compress the model on disk
+            :param skip_compression_stats: whether to skip the calculation of
+            compression statistics (such as global sparsity and sparsity structure) when
+            saving a model in dense format
+            :param kwargs: additional kwargs to pass on to model.save_pretrained
+            """
+            model = model_ref()
+
+            if sparsity_config is not None:
+                sparsity_config.fill_config_details(model)
+            elif not skip_compression_stats:
+                # try to infer a sparsity config from the model if none is provided
+                _LOGGER.info(
+                    "Inferring a sparsity configuration requires a global sparsity "
+                    "calculation. This can be costly for large models. To skip the "
+                    "calculation of compression statistics set "
+                    "skip_compression_stats=True"
+                )
+                sparsity_config = CompressionConfig.infer_config_from_model(
+                    model, compress=save_compressed
+                )
+
+            if sparsity_config is None:
+                # model is not sparse, save as dense
+                return original_save_pretrained.__get__(model, model_class)(
+                    save_directory, **kwargs
+                )
+
+            # if we've gotten to this point we have a config so we can run compression
+            kwargs["safe_serialization"] = True
```

Comment:
i guess this is where it doesn't matter about safetensors

Reply:
Compression is only set up to work with safetensors files, so we force it if we're running compression